osu.Game/Rulesets/Objects/Pooling/PoolableDrawableWithLifetime.cs

```diff
@@ -26,17 +27,18 @@ public abstract class PoolableDrawableWithLifetime<TEntry> : PoolableDrawable wh
         /// </summary>
         protected bool HasEntryApplied { get; private set; }
 
-        // Drawable's lifetime gets out of sync with entry's lifetime if entry's lifetime is modified.
-        // We cannot delegate getter to `Entry.LifetimeStart` because it is incompatible with `LifetimeManagementContainer` due to how lifetime change is detected.
         public override double LifetimeStart
         {
             get => base.LifetimeStart;
             set
             {
-                base.LifetimeStart = value;
+                if (Entry == null && LifetimeStart != value)
+                    throw new InvalidOperationException($"Cannot modify lifetime of {nameof(PoolableDrawableWithLifetime<TEntry>)} when entry is not set");
 
-                if (Entry != null)
-                    Entry.LifetimeStart = value;
+                if (Entry == null) return;
+
+                Entry.LifetimeStart = value;
+                base.LifetimeStart = Entry.LifetimeStart;
```

Comment:
Can use compound assignment:

```suggestion
                base.LifetimeStart = Entry.LifetimeStart = value;
```

Same goes for `End`, obviously.

Reply:
This transformation was incorrect when `KeepAlive` is true.

Reply:
How does that flag make my proposal incorrect? Because that sounds broken to hell. It's been cited twice now as a reason for writing really awkward code. And *if* it turns out that there is a *very* good reason for splitting the assignment, it better be commented.

Reply:
While `HitObjectLifetimeEntry.KeepAlive` is true, lifetime change is blocked and `Entry.LifetimeStart` will return `double.MinValue` rather than the value that is just assigned.
To sync the drawable lifetime and the entry lifetime, that value must be assigned to `base.LifetimeStart`.
If the compound assignment is used, it will be `base.LifetimeStart = value`.

Reply:
I'm confused. Assignment is right-associative, is it not? There should be no functional difference in execution.

That said I really don't like that the order matters here, but alas.

Reply:
Never mind, I get it. It still feels quite bad but can't see a way to avoid it.

Reply:
It turns out setting `base.LifetimeStart` is not required here because entry lifetime will cause lifetime change event and lifetime is synced in the callback.